R/get_coding_ssm.R

```diff
@@ -0,0 +1,153 @@
+#' @title Get Coding SSMs
+#'
+#' @description Convenience function for loading coding Simple Somatic Mutations (SSM) from the bundled data [GAMBLR.data::sample_data].
+#'
+#' @details This "bare bones" function was developed to retrieve coding SSM calls for non-GSC-users.
+#' For users with GSC access it is highly recommended to instead call [GAMBLR.results::get_coding_ssm].
+#' Effectively retrieve coding SSM calls. Multiple filtering parameters are available for this function.
+#' For more information on how to implement the filtering parameters, refer to the parameter descriptions as well as examples in the vignettes.
+#' It relies on the bundled sample data in this package. 
+#'
+#' @param limit_cohort Supply this to restrict mutations to one or more cohorts in a vector.
+#' @param exclude_cohort  Supply this to exclude mutations from one or more cohorts in a vector.
+#' @param limit_pathology Supply this to restrict mutations to one pathology.
+#' @param limit_samples Supply this to restrict mutations to a vector of sample_id (instead of subsetting using the provided metadata)
+#' @param these_samples_metadata Supply a metadata table to auto-subset the data to samples in that table before returning
+#' @param force_unmatched_samples Optional argument for forcing unmatched samples, using [GAMBLR.data::get_ssm_by_samples].
+#' @param projection Reference genome build for the coordinates in the MAF file. The default is hg19 genome build.
+#' @param seq_type The seq_type you want back, default is genome.
+#' @param basic_columns Set to FALSE to override the default behavior of returning only the first 45 columns of MAF data.
+#' @param maf_cols if basic_columns is set to FALSE, the user can specify what columns to be returned within the MAF. This parameter can either be a vector of indexes (integer) or a vector of characters (matching columns in MAF).
+#' @param from_flatfile This parameter does not do anything for this version of get_manta_sv. See [GAMBLR.results::get_coding_ssm] for more info.
+#' @param augmented This parameter does not do anything for this version of get_manta_sv. See [GAMBLR.results::get_coding_ssm] for more info.
+#' @param min_read_support Only returns variants with at least this many reads in t_alt_count (for cleaning up augmented MAFs)
+#' @param groups This parameter does not do anything for this version of get_manta_sv. See [GAMBLR.results::get_coding_ssm] for more info.
+#' @param include_silent Logical parameter indicating whether to include silent mutations into coding mutations. Default is TRUE.
+#' @param engine This parameter does not do anything for this version of get_manta_sv. See [GAMBLR.results::get_coding_ssm] for more info.
+#'
+#' @return A data frame (.maf) with coding SSMs in reference to the selected projection.
+#' 
+#' @export
+#' 
+#' @examples
+#' #return SSMs in reference to GRCh37:
+#' ssm_grch37 = get_coding_ssm(seq_type = "genome")
```

Comment:
Have a look at the output from this after you run the examples. Hopefully it's clear that there's a problem with this output given the intention of this function. 
```
table(ssm_grch37$Variant_Classification)
```

Reply:
Yes, this has been fixed. Thanks!